dask/array/tests/test_optimization.py

```diff
@@ -388,6 +388,23 @@ def test_disable_lowlevel_fusion():
         assert_eq(y, [1] * 3)
 
 
+def test_array_creation_blockwise_fusion():
+    """
+    Check that certain array creation routines work with blockwise and can be
+    fused with other blockwise operations.
+    """
+    x = da.ones(3, chunks=(3,))
+    y = da.zeros(3, chunks=(3,))
+    z = da.full(3, fill_value=2, chunks=(3,))
+    a = x + y + z
+    dsk1 = a.__dask_graph__()
+    assert len(dsk1) == 5
+    with dask.config.set({"optimization.fuse.active": False}):
+        dsk2 = optimize(dsk1, y.__dask_keys__())
```

Comment:
This is totally fine (you want high-level fusion, but need to disable low-level fusion to avoid raw-graph materialization).

It may make sense to use [optimize_blockwise](https://github.com/dask/dask/blob/1d669480f78efa290fad612d196ecd18d26e7972/dask/blockwise.py#L1050) directly (instead of `optimize` with fusion deactivated), since the goal here is to check that high-level fusion works.

Reply:
Makes sense